File: hash_table/0739_daily_temperatures/0739_daily_temperatures.go
package main 

import "fmt"

func dailyTemperatures(T []int) []int {
	res := make([]int, len(T), len(T))
	stack := make([]int, 1, len(T))

	for i := 1; i < len(T); i++ {
		for len(stack) > 0 && T[i] > T[stack[len(stack) - 1]] {
			idx := stack[len(stack)-1]
			res[idx] = i - idx
			stack = stack[:len(stack) - 1]

		}
		stack = append(stack, i)
	}

	return res
}

func main() {
	T := []int{73,74,75,71,69,72,76,73}
	res := dailyTemperatures(T)
	fmt.Println(res)
}

File: hash_table/0739_daily_temperatures/0739_daily_temperatures.py
class Solution(object):
    def dailyTemperatures(self, T):
        result = [0] * len(T)
        stk = []
        for i in xrange(len(T)):
            while stk and T[stk[-1]] < T[i]:
                idx = stk.pop()
                result[idx] = i - idx
            stk.append(i)
            # print(stk, result)
        return result


T = [73,74,75,71,69,72,76,73]

res = Solution().dailyTemperatures(T)
print(res)